### PR TITLE
Query objects section on code-conventions page

### PR DESCRIPTION
**Pull request by brentonannan**

### CODE REVIEW

File: code-conventions/README.md [hunk 1]
```diff
@@ -105,3 +106,60 @@ module Workflow
...
+  * Some combination of these
+* It may be prudent to ignore rules of DRY in some circumstances, e.g. reimplementing a scope that already exists on a model, so it is clear how it's specified *in the context of this query*, even if it may be the same as that on the model (for the time being)
+
+Example:
+
+```ruby
+module Query
+  module Subscription
+    module TeacherTrials
+      extend self
```
Comment (stuart) [inline code-conventions/README.md]: Rubocop complains about `extend self`, it recommends using `module_function foo` instead.
I think the reasoning is that you then only expose the functions you really want to be module functions rather 
than everything. Personally I don't care either way.

There are gotchas with module_function: http://blog.rubybestpractices.com/posts/gregory/041-issue-10.5-uses-for-modules.html

**Comment by brentonannan** [comment]
@martinstannard [this](https://github.com/blake-education/guides/pull/9/files#diff-2b6fbc4d0dadf4545506f3e7c9b79856R130) isn't concrete enough?

Edit: I didn't see this was a line comment on a commit, will respond there.

File: code-conventions/README.md (hunk 1 above)
Comment (brentonannan) [inline code-conventions/README.md]: I'd prefer to stop it complaining, but happy to go with the hive mind if we could get some more opinions on this.

Comment (stuart) [inline code-conventions/README.md]: We can stop it complaining by adjusting the rubocop config of course.

Comment (gogogarrett) [inline code-conventions/README.md]: My version of the rubocop actually doesn't mind this `extend self` - but I actually like the idea of `module_function` - so it's whatever.

Comment (novemberkilo) [inline code-conventions/README.md]: :+1: for using `module_function` over `extend self`

**Comment by novemberkilo** [comment]
@brentonannan looks great - thanks for putting this together - :sheep: it when consensus

**Comment by brentonannan** [comment]
Switched to use `module_function` @stuart, and moved the examples @martinstannard.

**Comment by kunzmann** [comment]
:+1:

**Comment by keithpitty** [comment]
Looks good, @brentonannan. Thanks for this.

**Comment by martinstannard** [comment]
thanks @brentonannan

**Pull request merged**